Plugins/Covfie/src/FieldConversion.cpp

```diff
@@ -0,0 +1,181 @@
+/// @param cache The cache.
+/// @param nBins 3D array of containing the number of bins for each axis.
+/// @param min (min_x, min_y, min_z)
+/// @param max (max_x, max_y, max_z)
+/// @return A clamp affine linear strided covfie field.
+template <typename magnetic_field_t, typename cache_t, typename point3_1_t,
+          typename point3_2_t, typename point3_3_t>
+interpolated_field_t covfieFieldLinear(const magnetic_field_t& magneticField,
+                                       cache_t& cache, const point3_1_t& nBins,
+                                       const point3_2_t& min,
+                                       const point3_3_t& max) {
+  auto builder = newBuilder(magneticField, cache, nBins, min, max);
+  interpolated_field_t::backend_t::configuration_t clampConfiguration{
+      {std::nextafter(static_cast<float>(min[0]),
+                      std::numeric_limits<float>::infinity()),
+       std::nextafter(static_cast<float>(min[1]),
+                      std::numeric_limits<float>::infinity()),
+       std::nextafter(static_cast<float>(min[2]),
+                      std::numeric_limits<float>::infinity())},
+      {std::nextafter(static_cast<float>(max[0]),
+                      -std::numeric_limits<float>::infinity()),
+       std::nextafter(static_cast<float>(max[1]),
+                      -std::numeric_limits<float>::infinity()),
+       std::nextafter(static_cast<float>(max[2]),
+                      -std::numeric_limits<float>::infinity())}};
+
+  interpolated_field_t field(covfie::make_parameter_pack(
+      std::move(clampConfiguration),
+      affineConfiguration<interpolated_field_t::backend_t::backend_t>(nBins,
+                                                                      min, max),
+      interpolated_field_t::backend_t::backend_t::backend_t::configuration_t{},
+      builder.backend()));
+
+  return field;
+}
+
+/// @brief Creates a covfie field from a magnetic field provider by sampling it.
+/// @param magneticField The acts magnetic field provider.
+/// @param cache The acts cache.
+/// @param nBins 3D array of containing the number of bins for each axis.
+/// @param min (min_x, min_y, min_z)
+/// @param max (max_x, max_y, max_z)
+/// @return A clamp affine linear strided covfie field.
+interpolated_field_t covfieField(
+    const Acts::MagneticFieldProvider& magneticField,
+    Acts::MagneticFieldProvider::Cache& cache,
+    const std::vector<std::size_t>& nBins, const std::vector<double>& min,
+    const std::vector<double>& max) {
+  return covfieFieldLinear(magneticField, cache, nBins, min, max);
+}
+
+/// @brief Creates a covfie field from an interpolated magnetic field.
+/// @param magneticField The acts interpolated magnetic field.
+/// @return A clamp affine linear strided covfie field.
+interpolated_field_t covfieField(
+    const Acts::InterpolatedMagneticField& magneticField) {
+  Acts::MagneticFieldContext ctx;
+  auto cache = magneticField.makeCache(ctx);
```

Comment:
This object is a type-erased concrete type, allowing you to remove the templating on the cache type: It's always `MagneticFieldProvider::Cache`.